astropy/io/votable/tests/converter_test.py

```diff
@@ -48,14 +47,13 @@ def test_char_mask():
 
 def test_oversize_unicode():
     config = {'verify': 'exception'}
-    with catch_warnings(exceptions.W46) as w:
+    with pytest.raises(exceptions.W47):
```

Comment:
@tomdonaldson , I don't understand what is going on here but I had to make thing change to make the test pass changing from `catch_warnings` to `pytest`. Does this make sense to you?

Reply:
Interesting... when I ran this manually (see below), both W46 and W47 are emitted but at different lines. I will modify this to reflect the reality that I see. I think when you pass in `catch_warnings(exceptions.W46)`, it only cares about that warning and ignores everything else silently. So I think what the old code was really testing was "make sure there is only one W46" emitted. I am still not sure why I needed `pytest.raises`, I'll dig around more.
```python
>>> from astropy.io.votable import converters, tree                                             
>>> config = {'verify': 'exception'}                                                            
>>> field = tree.Field(None, name='c2', datatype='unicodeChar', config=config)                  
WARNING: W47: ?:?:?: W47: Missing arraysize indicates length 1 [astropy.io.votable.converters]
>>> c = converters.get_converter(field, config=config)                                          
>>> c.parse("XXX")                                                                              
WARNING: W46: ?:?:?: W46: unicodeChar value is too long for specified length of 1 [astropy.io.votable.converters]
('XXX', False)
```